firestore/src/common/load_bundle_task_progress.cc

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "firestore/src/include/firebase/firestore/load_bundle_task_progress.h"
+
+#include <cstdint>
+
+#include "app/src/assert.h"
+#if defined(__ANDROID__)
+#include "firestore/src/android/load_bundle_task_progress_android.h"
+#else
+#include "firestore/src/ios/load_bundle_task_progress_ios.h"
+#endif  // defined(__ANDROID__)
+
+namespace firebase {
+namespace firestore {
+
+LoadBundleTaskProgress::LoadBundleTaskProgress(int32_t documents_loaded,
+                                               int32_t total_documents,
+                                               int64_t bytes_loaded,
+                                               int64_t total_bytes, State state)
+    : documents_loaded_(documents_loaded),
+      total_documents_(total_documents),
+      bytes_loaded_(bytes_loaded),
+      total_bytes_(total_bytes),
+      state_(state) {}
+
+LoadBundleTaskProgress::LoadBundleTaskProgress(
+    LoadBundleTaskProgressInternal* internal) {
+  FIREBASE_ASSERT(internal != nullptr);
+  documents_loaded_ = internal->documents_loaded();
+  total_documents_ = internal->total_documents();
+  bytes_loaded_ = internal->bytes_loaded();
+  total_bytes_ = internal->total_bytes();
+  state_ = internal->state();
+
+  delete internal;
```

Comment:
Is this conventional where the constructor deletes the `internal` object? This looks very risky. Is there a better way to achieve this?